ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java

```diff
@@ -481,23 +486,17 @@ public void removeImportedTransaction(Xid xid) throws XAException {
         SubordinationManager.getTransactionImporter().removeImportedTransaction(xid);
     }
 
-    public Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, String parentNodeName, int recoveryFlags) throws XAException {
+    public Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, String parentNodeName, int recoveryFlags) throws XAException, UnsupportedOperationException {
```

Comment:
You have changed the signature of the interface ExtendedJBossXATerminator#getXidsToRecoverForParentNode and we normally deprecate first. What I would suggest is that we do not declare unchecked exceptions in the interface since the caller is not meant to handle them - add it to the javadoc instead.